target_api/middlewares.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth.models import AnonymousUser
+
+from channels.auth import AuthMiddlewareStack
+from channels.db import database_sync_to_async
+from rest_framework.authtoken.models import Token
+
+
+@database_sync_to_async
+def get_user_by_query_param(name, value):
+    try:
+        if name.lower() == 'token':
+            token = Token.objects.get(key=value)
+            user = token.user
+    except Token.DoesNotExist:
+        pass
+    return user or AnonymousUser()
```

Comment:
Can `name.lower()` be other than `'token'`? I think that in that case this will throw an error because in the return statement `user` is not a defined variable.

Reply:
Yes, I agree with that improvement. 
What about if I raise an exception when the name is not `token`, then catch it and return `AnonymousUser`? So, I only return `user` when everything is ok. wdyt?